ec2/ec2test/internet_gateways_test.go

```diff
@@ -0,0 +1,112 @@
+//
+// goamz - Go packages to interact with the Amazon Web Services.
+//
+//   https://wiki.ubuntu.com/goamz
+//
+// Copyright (c) 2015 Canonical Ltd.
+//
+
+package ec2test_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"gopkg.in/amz.v3/ec2"
+)
+
+func (s *S) TestAddInternetGateway(c *C) {
+	igw, err := s.srv.AddInternetGateway(ec2.InternetGateway{
+		VPCId: "missing",
+	})
+	c.Assert(err, ErrorMatches, `VPC "missing" not found`)
+	c.Assert(igw, DeepEquals, ec2.InternetGateway{})
+	s.assertVPCsExist(c, true)
+
+	// Added as not attached to a VPC when VPCId is empty.
+	toAdd := ec2.InternetGateway{
+		Id:              "should-be-ignored",
+		VPCId:           "",
+		AttachmentState: "insane",
+	}
+	added, err := s.srv.AddInternetGateway(toAdd)
+	c.Assert(err, IsNil)
+	c.Assert(added.Id, Matches, `^igw-[0-9a-f]+$`)
+	toAdd.Id = added.Id // the only field that should differ
+	c.Assert(added, DeepEquals, toAdd)
+
+	filter := ec2.NewFilter()
+	filter.Add("attachment.state", "insane")
+	resp, err := s.ec2.InternetGateways(nil, filter)
+	c.Assert(err, IsNil)
+	c.Assert(resp, NotNil)
+	c.Assert(resp.InternetGateways, HasLen, 1)
+	c.Assert(resp.InternetGateways[0], DeepEquals, added)
+
+	// Added as attached to a VPC when VPCId is valid.
+	toAdd = ec2.InternetGateway{
+		Id:              "should-be-ignored",
+		VPCId:           "vpc-0", // testing default VPC
+		AttachmentState: "available",
+	}
+	added, err = s.srv.AddInternetGateway(toAdd)
+	c.Assert(err, IsNil)
+	c.Assert(added.Id, Matches, `^igw-[0-9a-f]+$`)
+	toAdd.Id = added.Id // the only field that should differ
+	c.Assert(added, DeepEquals, toAdd)
+
+	resp, err = s.ec2.InternetGateways([]string{added.Id}, nil)
+	c.Assert(err, IsNil)
+	c.Assert(resp, NotNil)
+	c.Assert(resp.InternetGateways, HasLen, 1)
+	c.Assert(resp.InternetGateways[0], DeepEquals, added)
+}
+
+func (s *S) TestUpdateInternetGateway(c *C) {
+	// AddDefaultVPCAndSubnets() creates a IGW as well.
```

Comment:
I'll fix the typo. AddDefaultVPCAndSubnets is called in SetUpTest of the S suite.